delira/io/tf.py

```diff
@@ -35,3 +37,55 @@ def load_checkpoint(file: str, model=None):
     # this for memory leak
     tf.train.Saver().restore(model._sess, file)
     return {}
+
+
+def _create_varlist(model: AbstractTfEagerNetwork = None,
+                    optimizer: typing.Dict[str, tf.train.Optimizer] = None):
+    variable_list = []
+
+    if model is not None:
+        variable_list += model.variables
+
+    if optimizer is not None:
+        for k, v in optimizer.items():
+            variable_list += v.variables()
+
+    return variable_list
+
+
+def save_checkpoint_eager(file,
```

Comment:
While this works for now, I suggest switching to [tf.train.Checkpoint](https://www.tensorflow.org/api_docs/python/tf/train/Checkpoint) for both Eager and Graph mode in future.

Reply:
tracked in #178